chore: cut 0.77.0 (#6950)
prometheus-operator/prometheus-operator: CHANGELOG.md

## Unreleased

## 0.77.0 / 2024-09-19

> > Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.

Reviewer: Nothing required, but would be nice if we callouts: https://github.com/orgs/community/discussions/16925

info and warning wdyt?

Author: This is nice. I had thought about updating our README as well when I saw something similar in metrics-server repo.

I will update. Thanks for the suggestion 🙂

Author: https://github.com/prometheus-operator/prometheus-operator/blob/0a82836229afad0c1044dfcc0d03d3c8fb13f15f/CHANGELOG.md looks nice :D

## Unreleased

## 0.77.0 / 2024-09-18

> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.
> This project is currently under active development and fully experimental, so expect breaking changes and rough edges. We encourage you to [try it out](https://github.com/prometheus-operator/poctl/releases/tag/v0.1.0) and provide the feedback.

> * [CHANGE] Add more API validation and changes in `ScrapeConfig CRD`. #6921 #6855 #6744

Reviewer 2: ```suggestion
* [CHANGE] Add more API validation and changes to the `ScrapeConfig` CRD. #6921 #6855 #6744
```

* [CHANGE] Update the validating webhook to reject `PrometheusRule` objects that would generate configuration greater than the max ConfigMap size (512KiB). #6606

> * [FEATURE] Add experimental Prometheus 3 support. This will allow users to try new Prometheus `v3.0.0-beta.0`. #6940

Reviewer 2: ```suggestion
* [FEATURE] Add experimental Prometheus 3 support. It allows to try the new Prometheus `v3.0.0-beta.0`. #6940
```

  * Feature requests in this version won't be supported till Prometheus 3 is stable but this would let users to try the beta version with Operator and report issues if any.

> * [FEATURE] Allow Prometheus Agent to update, delete, and handle changes to DaemonSet object. #6787

Reviewer 2: after more thoughts, I'm not sure that we need to include this in the changelog.

  * Feature requests in this version won't be supported till Prometheus 3 is stable but this would let users to try the beta version with Operator and report issues if any.
* [FEATURE] Add support for `PodMonitor` to the `PrometheusAgent` CRD in DaemonSet mode. #6879
* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller. #6882
* [FEATURE] Add `outOfOrderTimeWindow` field to the `PrometheusAgent` CRD. #6851
* [FEATURE] Add `matchFirstNetwork` field to the `ScrapeConfig` CRD for the Docker Service Discovery. #6849 #6894
* [FEATURE] Add the `otlp` field to the `Prometheus` and `PrometheusAgent` CRDs. #6780
* [FEATURE] Add support for certificate, key and client CA file paths to the web TLS configuration. #6358
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for OAuth2 clients. #6922
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for HTTP clients. #6864
* [FEATURE] Add Ionos Service Discovery support to `ScrapeConfig` CRD. (#6732)

> * [BUGFIX] Fix panic when processing an invalid `AlertmanagerConfig` object used for global configuration. #6931

Reviewer 2: can you remove L32?

Author: done

## Unreleased

## 0.77.0 / 2024-09-18

> 

Reviewer: I think we can remove this extra line

Author: we need blank line after heading for markdown

> Along with this release we also introduce a new command line tool [poctl](https://github.com/prometheus-operator/poctl). It is designed specifically for managing Prometheus Operator Custom Resources.

> 

Reviewer 2: remove this line to avoid a breaking line?

> This project is currently under active development and fully experimental, so expect breaking changes and rough edges. We encourage you to [try it out](https://github.com/prometheus-operator/poctl/releases/tag/v0.1.0) and provide the feedback.

* [CHANGE] Add more API validation and changes in ScrapeConfig CRD. #6921 #6855

> * [CHANGE] Update API validations for ECSDConfig in ScrapeConfig CRD. #6744

Reviewer 2: IMHO either we have a catch-all entry for ScrapeConfig CRD or we have one line per PR.

> * [FEATURE] Add experimental Prometheus 3 support. #6940

Reviewer: I was wondering if also makes sense highlight Prometheus 3 is experimental and we're not supporting it, wdyt?

* [FEATURE] Allow Prometheus Agent to update, delete, and handle changes to DaemonSet object. #6787

> * [FEATURE] Add support PodMonitor for Prometheus Agent in DaemonSet mode. #6879

Reviewer 2: ```suggestion
* [FEATURE] Add support for `PodMonitor` to the `PrometheusAgent` CRD in DaemonSet mode. #6879
```

* [FEATURE] Add `-kubelet-endpointslice` argument to support `EndpointSlice` for the kubelet controller. #6882

> * [FEATURE] Reject PrometheusRule objects that generate > 512KiB configmap. #6606

Reviewer 2: ```suggestion
* [CHANGE] Update the validating webhook to reject `PrometheusRule` objects that would generate configuration greater than the max ConfigMap size (512KiB). #6606
```

> * [FEATURE] Add support for `OutOfOrderTimeWindow` field for PrometheusAgent. #6851

Reviewer 2: ```suggestion
* [FEATURE] Add `outOfOrderTimeWindow` field to the `PrometheusAgent` CRD. #6851
```

> * [FEATURE] Add support for `MatchFirstNetwork` field for DockerSD. #6849 #6894

Reviewer 2: ```suggestion
* [FEATURE] Add `matchFirstNetwork` field to the `ScrapeConfig` CRD for the Docker Service Discovery. #6849 #6894
```

> * [FEATURE] Add support for new otlp receiver configuration. #6780

Reviewer 2: ```suggestion
* [FEATURE] Add the `otlp` field to the `Prometheus` and `PrometheusAgent` CRDs. #6780
```

> * [FEATURE] Add support for file paths for certificate, key and client CA in web TLS. #6358

Reviewer 2: ```suggestion
* [FEATURE] Add support for certificate, key and client CA file paths to the web TLS configuration. #6358
```

> * [FEATURE] Add new proxy fields into the AlertmanagerConfig CRD for the OAuth2 clients. #6922

Reviewer 2: ```suggestion
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for OAuth2 clients. #6922
```

> * [FEATURE] Add ProxyConfig fields into the AlertmanagerConfig CRD for the HTTP clients. #6864

Reviewer 2: ```suggestion
* [FEATURE] Add `noProxy`, `proxyFromEnvironment` and `proxyConnectHeader` fields to the AlertmanagerConfig CRD for HTTP clients. #6864
```

> * [FEATURE] Add Ionos Service Discovery support to ScrapeConfig CRD. (#6732)

Reviewer 2: ```suggestion
* [FEATURE] Add Ionos Service Discovery support to `ScrapeConfig` CRD. (#6732)
```

> * [BUGFIX] Fix AlertmanagerConfig objects with invalid child routes definition to cause problems. #6931

Reviewer 2: I added it to the CHANGELOG.md already but at a wrong place :facepalm: 
https://github.com/prometheus-operator/prometheus-operator/blob/c846674a17c6db9d1bd04d08300c354f960d909b/CHANGELOG.md?plain=1#L11

> * [BUGFIX] Fix avoid BADKEY error in logs. #6936

Reviewer 2: We can omit this one because it wasn't part of an official release.

> * [BUGFIX] Allow proxy URL for all Prometheus versions. #6845

Reviewer 2: ```suggestion
* [BUGFIX] Allow configuring proxy URL for all Prometheus versions. #6845
```
